external/infrastructure/aws-ec2/main.tf

```diff
@@ -0,0 +1,187 @@
+# --------------------------------------------
+# Provider and credentials
+# --------------------------------------------
+provider "aws" {
+  version = "~> 2.0"
+  region = var.region
+  secret_key = var.secret-key
+  access_key = var.access-key
+}
+
+# --------------------------------------------
+# Upload ssh public key on machine
+# --------------------------------------------
+resource "aws_key_pair" "dejaq" {
+  key_name = "dejaq"
+  public_key = var.public-key
+  tags = {
+    project-dejaq = "dejaq-test"
```

Comment:
project should be generic ("project" not "project-dejaq") as you can have multiple projects in AWS only one is dejaq